### PR TITLE
Remove upgrade, remove --force-confnew, add docker flags

### PR DESCRIPTION
**Pull request by ostacey**
Removing the upgrade fixes added in Version 2.1. These were determined to not be useful.
While they were able to repair two issues we saw when upgrading Docker from 1.5 to 1.11,
there was a third issue that just destroyed some data containers on update. So it was
determined that the only safe way to update was to move all data out of containers before
the upgrade and recreate all containers post-upgrade. With that in mind, it became clear
that this code was adding complexity and no value here, so it was removed.

We also no longer pass the "--force-confnew" flag to apt when installing the Docker package.
This flag, added in 2.1.2, turned out to have some undesirable side effects - it meant that
any Docker config written before the Docker install would be erased during install, which
in turn meant we'd always have to start Docker with the wrong config, then immediately stop
it, fix the config, and start it again - not good.

This role now manages the "/etc/default/docker" file. This was done to allow us to set
Docker configuration that would be applied before startup.

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -62,17 +62,30 @@
...
+- name: Restart docker on config change
+  service: name=docker state=restarted
+  when: r_etc_default_docker.changed and not r_docker_package_install.changed
 
 - name: Ensure that Docker is running
   service: name="docker" state=started
 
+- name: reread docker facts
+  setup: filter=ansible_docker0
+  when: r_etc_default_docker|changed or r_docker_package_install|changed
```
Comment (charness) [inline tasks/main.yml:87]: Why the `|changed` filter here, but `.changed` on the same variables 7 lines earlier?

Comment (ostacey) [reply tasks/main.yml:87]: Good catch, will flatten.

File: tasks/docker-engine.yml [hunk 2]
```diff
@@ -19,74 +19,8 @@
...
-# processes for the containers that existed before the upgrade aren't stopped by the upgrade, but
-# instead are left hanging around with parent PID 1. Shutting down docker before the upgrade seems
-# to prevent this.
-- name: shut down docker before upgrade
-  service: name=docker state=stopped
-  when: "{{ docker_exec_stat_result.stat.exists and installed_docker_version != docker_version }}"
- 
 - name: Install Docker Engine
   apt:
     name: "docker-engine={{ docker_version }}-0~{{ ansible_distribution_release }}"
```
Comment (yarloocll) [inline tasks/docker-engine.yml]: probably don't really need to change, but i've seen some packages that have a different debian revision on different ubuntu releases (so trusty has one debian revision and percise has another) and i think we probably want the latest version for the release.

It is probably a rare case, but it is possible. not sure we want to parameterize this... but might save time if it ever comes up.

https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-Version
https://docs.docker.com/engine/installation/linux/ubuntulinux/#/update-your-apt-sources (item 8, has some information about versions)

Comment (ostacey) [reply tasks/docker-engine.yml]: You're right, that could be an issue. It's easy enough to include.

Comment (ostacey) [reply tasks/docker-engine.yml]: ... see update.

**Pull request merged**